derive/src/options.rs

```diff
@@ -1,37 +1,63 @@
-use proc_macro::TokenStream;
+use std::str::FromStr;
+
+use crate::params_style::ParamStyle;
+use crate::rpc_attr::path_eq_str;
+
+const CLIENT_META_WORD: &str = "client";
+const SERVER_META_WORD: &str = "server";
+const PARAMS_META_KEY: &str = "params";
 
 #[derive(Debug)]
 pub struct DeriveOptions {
 	pub enable_client: bool,
 	pub enable_server: bool,
+	pub params_style: ParamStyle,
 }
 
 impl DeriveOptions {
-	pub fn new(enable_client: bool, enable_server: bool) -> Self {
+	pub fn new(enable_client: bool, enable_server: bool, params_style: ParamStyle) -> Self {
 		DeriveOptions {
 			enable_client,
 			enable_server,
+			params_style,
 		}
 	}
 
-	pub fn try_from(tokens: TokenStream) -> Result<Self, syn::Error> {
-		if tokens.is_empty() {
-			return Ok(Self::new(true, true));
-		}
-		let ident: syn::Ident = syn::parse::<syn::Ident>(tokens)?;
-		let options = {
-			let ident = ident.to_string();
-			if ident == "client" {
-				Some(Self::new(true, false))
-			} else if ident == "server" {
-				Some(Self::new(false, true))
-			} else {
-				None
+	pub fn try_from(args: syn::AttributeArgs) -> Result<Self, syn::Error> {
+		let mut options = DeriveOptions::new(false, false, ParamStyle::default());
+		for arg in args {
+			if let syn::NestedMeta::Meta(meta) = arg {
+				match meta {
+					syn::Meta::Path(p) => {
+						match p.get_ident().unwrap().to_string().as_ref() {
```

Comment:
Can this `unwrap` fail? Can we return an error instead?

Reply:
Throws a nice compiler error now instead